templates/assets/plugins/sequencer/popcorn.sequencer.js

```
@@ -0,0 +1,506 @@
+    quarantine.style.height = "0px";
+    quarantine.style.overflow = "hidden";
+    quarantine.style.visibility = "hidden";
+    document.body.appendChild( quarantine );
+
+    getIframeQuarantine.instance = quarantine;
+    return quarantine;
+  }
+
+  var MEDIA_LOAD_TIMEOUT = 10000;
+ 
+  Popcorn.plugin( "sequencer", {
+    _setup: function( options ) {
+      var _this = this;
+
+      options.setupContainer = function() {
+        var container = document.createElement( "div" ),
+            target = Popcorn.dom.find( options.target );
+
+        if ( !target ) {
+          target = _this.media.parentNode;
+        }
+
+        options._target = target;
+        options._container = container;
+
+        container.style.zIndex = 0;
+        container.className = "popcorn-sequencer";
+        container.style.position = "absolute";
+        container.style.width = "100%";
+        container.style.height = "100%";
+        container.style.top = 0;
+        container.style.left = 0;
+
+        if ( target ) {
+          target.appendChild( container );
+        }
+      };
+      options.displayLoading = function() {
+        document.querySelector( ".loading-message" ).classList.add( "show-media" );
+      };
+      options.hideLoading = function() {
+        document.querySelector( ".loading-message" ).classList.remove( "show-media" );
+      };
+      if ( !options.from || options.from > options.duration ) {
+        options.from = 0;
+      }
+
+      options.readyEvent = function() {
+        options.failed = false;
+        options.p.off( "loadedmetadata", options.readyEvent );
+        options.ready = true;
+        options._container.style.width = ( options.width || "100" ) + "%";
+        options._container.style.height = ( options.height || "100" ) + "%";
+        options._container.style.top = ( options.top || "0" ) + "%";
+        options._container.style.left = ( options.left || "0" ) + "%";
+        _this.on( "volumechange", options._volumeEvent );
```

Could `options` ever lack an `_volumeEvent` function?  Should you guard here?  I guess at worst it will silently fail, but that kind of sucks, and we seem to have various volume related bugs.